hibernate-core/src/main/java/org/hibernate/proxy/pojo/javassist/JavassistLazyInitializer.java

```diff
@@ -57,6 +57,11 @@ public Object invoke(
 			final Method proceed,
 			final Object[] args) throws Throwable {
 		if ( this.constructed ) {
```

Comment:
When debugging the [test case](https://github.com/hibernate/hibernate-orm/pull/1480/commits/b678a8f683de34b4b77fc39e5c149c63f1c18983) for this PR, I can see that `thisMethod` and `proceed` arguments are both `...$$_hibernate_getEntityEntry`, and so it works without initializing the proxy.

I'm not sure how `thisMethod` and `proceed` and are determined in general (ie., are they always the same when the method is prefixed "$$_hibernate_getEntityEntry"?), so I'm not sure if executing `proceed` is always the correct action.

@sebersole WDYT?